lib/hanami/routing/route.rb

```diff
@@ -50,7 +50,8 @@ def generate(resolver, options = {}, &endpoint)
       # @since 0.2.0
       # @api private
       def nested_router
-        dest.routes if dest.respond_to?(:routes)
+        dest.routes if dest.respond_to?(:routes) &&
```

Comment:
One thing to note here is that the `dest.respond_to?(:routes) &&` is possibly no longer needed as it was a less ideal way of identifying Hanami::Router instances before. If you agree this is no longer needed I will remove it.